src/ir/types/schema.h

```diff
@@ -22,6 +22,39 @@ class Schema {
 
   friend class TypeFactory;
 
+  // return a string representing the Schema in the following format:
+  //  schema <schema name> {
+  //     <field_name>: <field_kind>
+  //     ...
+  //  }
+  // only showing the first layer of the schema, e.g. not returning the values
+  // of <field_kind> for easier testing, fields are sorted alphabetically based
+  // on their name
+  std::string ToString() const {
+    std::string schema_name = name_.value_or("");
+    std::string schema_to_string = absl::StrCat(
+        "schema ", schema_name, schema_name == "" ? "{\n" : " {\n");
+    std::vector<std::string> fields_to_string;
+    for (const auto& [name, kind] : fields_) {
+      auto kind_to_string =
+          kind.type_base().kind() ==
+                  raksha::ir::types::TypeBase::Kind::kPrimitive
+              ? "primitive"
+              : "entity";
```

Comment:
Let us print the schema name instead of just a generic "entity"?

Reply:
What should we do when there is no schema name? Should the output be `entity <name>` or just `<name>`?

Reply:
when there is a name: just `<name>`
when there is no name, let us some constant say: "Entity".

If we fix #585, this would be moot anyway? 

In any case, let us defer this change to a later PR.

Reply:
I was working on this new PR already. I'll mark this as resolved and we can discuss elsewhere.